### PR TITLE
Use a real init process inside docker instead of runuser

### PR DESCRIPTION
**Pull request by quantum5**
`runuser` doesn't do the job of an `init` process such as reaping zombies.

### CODE REVIEW

**Comment by Riolku** [comment]
I missed the discussions out of band, what's the concern with reaping zombies?

**Comment by quantum5** [comment]
@Riolku go read https://stackoverflow.com/questions/49162358/docker-init-zombies-why-does-it-matter

File: .docker/tier1/Dockerfile [hunk 1]
```diff
@@ -11,4 +11,4 @@ RUN mkdir /judge /problems && cd /judge && \
 	runuser -u judge -w PATH -- /env/bin/dmoj-autoconf -V > /judge-runtime-paths.yml && \
 	echo '  crt_x86_in_lib32: true' >> /judge-runtime-paths.yml
 
-ENTRYPOINT ["/judge/.docker/entry"]
+ENTRYPOINT ["/usr/bin/tini", "--", "/judge/.docker/entry"]
```
Comment (Riolku) [inline .docker/tier1/Dockerfile:14]: What does `--` add?

Comment (Riolku) [reply .docker/tier1/Dockerfile:14]: Like, specifically here: I know what it does in general.

Comment (Riolku) [reply .docker/tier1/Dockerfile:14]: Oh it's for arguments that get passed on the command line on docker run.

**Review by Riolku — APPROVED** [review]

**Review by kiritofeng — APPROVED** [review]
LGTM

**Pull request merged**